src/templates.rs

```diff
@@ -275,6 +276,83 @@ pub fn register_case_helpers<'a>(mut h: Handlebars<'a>) -> Handlebars<'a> {
     h
 }
 
+#[derive(Clone, Copy)]
+pub struct FilterHelper;
+
+/// A Handlebars helper to filter an iterable JSON value.
+/// It receives the value to be filtered and a string containing the condition predicate,
+/// then uses Handlebars' truthy logic to filter the items in the value.
+/// It also supports the `#if` helper's `includeZero` optional parameter.
+impl HelperDef for FilterHelper {
+    fn call_inner<'reg: 'rc, 'rc>(
+        &self,
+        h: &Helper<'reg, 'rc>,
+        r: &'reg Handlebars<'reg>,
+        _ctx: &'rc Context,
+        _rc: &mut RenderContext<'reg, 'rc>,
+    ) -> Result<ScopedJson<'reg, 'rc>, RenderError> {
+        let value_param = h
+            .param(0)
+            .ok_or(RenderError::new("Filter helper: Param not found for index 0; must be value to be filtered"))?;
+
+        let value = value_param.value();
+
+        let condition_param = h
+            .param(1)
+            .ok_or(RenderError::new("Filter helper: Param not found for index 1; must be string containing filter condition predicate"))?;
+
+        let condition = condition_param
+            .value()
+            .as_str()
+            .ok_or(RenderError::new("Filter helper: filter condition predicate must be a string"))?;
```

Comment:
This is more of an aesthetic suggestion to keep code clean and prefer iterables

```suggestion
        let mut params = h.params().iter();
        let value = params
            .next()
            .ok_or(RenderError::new(
                "Filter helper: Param not found for index 0; must be value to be filtered",
            ))?
            .value();

        let condition = params
            .next()
            .ok_or(RenderError::new("Filter helper: Param not found for index 1; must be string containing filter condition predicate"))?
            .value()
            .as_str()
            .ok_or(RenderError::new("Filter helper: filter condition predicate must be a string"))?;
```